internal/controller/stages/stages.go

```diff
@@ -328,7 +328,10 @@ func (r *reconciler) syncStage(
 				Status: kargoapi.HealthStateHealthy,
 			}
 		}
-		status.CurrentFreight.Health = &health
+		status.Health = &health
+		if health.Status == kargoapi.HealthStateHealthy {
+			status.CurrentFreight.Qualified = true
+		}
```

Comment:
I think we shouldn't set health on Stages that don't have Argo CD App association. Not all stages in the UI should have hearts, such as the ones which are used for control-flow

Reply:
Totally agree. The PR with the new `NotApplicable` state would have addressed that, but I ended up closing it.

Especially since the structure of Health is going to change completely when I work on #570, I will get this for now to where Health is simply nil if not applicable.